### PR TITLE
null pointer exception

### PR DESCRIPTION
**Pull request by foxyriver**
**What this PR does / why we need it**:

> if len(mediaType) == 0 && len(mediaTypes) > 0 {
> 		mediaType = mediaTypes[0].MediaType
> 	}

    
mediaTypes[0] suggests to take the first element of the array mediaTypes, len(mediaTypes)>0 make sure that the first element is existing, avoid runtime error: index out of range.

**Special notes for your reviewer**:

**Release note**:

```release-note
```

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37260)

### CODE REVIEW

**Comment by dims** [comment]
@k8s-bot ok to test

**Comment by ncdc** [comment]
Could you please make your commit comment and PR description more descriptive about what you're doing here? Thanks!

**Comment by foxyriver** [comment]
@ncdc the PR description has been renewed, if still has other defects, please @ me.thanks

File: pkg/genericapiserver/api/handlers/negotiation/negotiate.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ func NegotiateOutputStreamSerializer(req *http.Request, ns runtime.NegotiatedSer
 func NegotiateInputSerializer(req *http.Request, ns runtime.NegotiatedSerializer) (runtime.SerializerInfo, error) {
 	mediaTypes := ns.SupportedMediaTypes()
 	mediaType := req.Header.Get("Content-Type")
-	if len(mediaType) == 0 {
+	if len(mediaType) == 0 && len(mediaTypes) > 0 {
```
Comment (lavalamp) [inline pkg/genericapiserver/api/handlers/negotiation/negotiate.go]: It should be a construction-time error to make a serializer without any mediaTypes, I think. Silently failing here is probably not great. @smarterclayton can confirm.

**Comment by lavalamp** [comment]
Sorry for delay, I didn't see this before I went OOO last month.

File: pkg/genericapiserver/api/handlers/negotiation/negotiate.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/genericapiserver/api/handlers/negotiation/negotiate.go]: Yeah, a serializer with no media types is not valid

**Comment by foxyriver** [comment]
@lavalamp @smarterclayton thx for your answer

**Pull request closed without merging**